src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerSearchService.kt

```diff
@@ -1,13 +1,25 @@
 package uk.gov.justice.hmpps.prison.service
 
+import jakarta.transaction.Transactional
 import org.springframework.stereotype.Component
+import uk.gov.justice.hmpps.prison.api.model.InmateDetail
 import uk.gov.justice.hmpps.prison.api.model.PrisonerSearchDetails
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderRepository
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderTransformer
+import java.util.Optional
 
 @Component
-class PrisonerSearchService(private val inmateService: InmateService) {
+class PrisonerSearchService(
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val offenderRepository: OffenderRepository,
+  private val offenderTransformer: OffenderTransformer,
+  private val inmateService: InmateService,
+) {
 
+  @Transactional
   fun getPrisonerDetails(offenderNo: String): PrisonerSearchDetails =
-    inmateService.findOffender(offenderNo, true, false)
```

Comment:
The main change is to stop using this service and load the offender and booking from JPA based repositories.